src/Kerbero.Infrastructure/NukiActions/Repositories/NukiSmartLockExternalRepository.cs

```diff
@@ -0,0 +1,49 @@
+using System.Net;
+using FluentResults;
+using Flurl;
+using Flurl.Http;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.Common.Models.ExternalResponses;
+using Kerbero.Domain.NukiActions.Models.ExternalResponses;
+using Kerbero.Domain.NukiActions.Repositories;
+using Kerbero.Domain.NukiAuthentication.Holders;
+using Kerbero.Infrastructure.Common.Extensions;
+using Kerbero.Infrastructure.Common.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Kerbero.Infrastructure.NukiActions.Repositories;
+
+public class NukiSmartLockExternalRepository: INukiSmartLockExternalRepository, ITokenAuthenticationHolder
+{
+	private readonly NukiExternalOptions _options;
+	private readonly ILogger<NukiSmartLockExternalRepository> _logger;
+	
+	public bool IsAuthenticated => !string.IsNullOrEmpty(Token);
+
+	public NukiSmartLockExternalRepository(IOptions<NukiExternalOptions> options, 
+		ILogger<NukiSmartLockExternalRepository> logger)
+	{
+		_options = options.Value;
+		_logger = logger;
+	}
+	
+	public async Task<Result<List<NukiSmartLockExternalResponse>>> GetNukiSmartLockList()
+	{
+		if (!IsAuthenticated || Token is null)
```

Comment:
```suggestion
		if (!IsAuthenticated)
```
credo che controlli gia null su IsAuthenticated